### PR TITLE
build-style/cmake: set CMAKE_BUILD_TYPE to None

### PR DESCRIPTION
**Pull request by sgn**
- I'm going to build all packages with `build_style=cmake` and fix them one by one before merging this in.

[ci skip]

### CODE REVIEW

**Comment by sgn** [comment]
Below changes will be submited separately:
- paraview: update to 5.9.0.
- vtk: devendor most of dependencies
- opencv4: update to 4.5.1.
- opencv: update to 3.4.13.
- libechonest-qt5: upstream moved
- k2pdfopt: update to 2.53.

File: srcpkgs/embree/template [hunk 1]
```diff
@@ -4,6 +4,7 @@ version=2.17.7
 revision=2
 archs="x86_64"
 build_style=cmake
+configure_args="-DTBB_ROOT=/usr -DEMBREE_IGNORE_CMAKE_CXX_FLAGS=OFF"
```
Comment (sgn) [inline srcpkgs/embree/template:7]: @thypon

Comment (thypon) [reply srcpkgs/embree/template:7]: I think it derived from archlinux package.

It should be defined this way now '-DEMBREE_MAX_ISA="AVX512SKX"'

**Comment by thypon** [comment]
What's the reason for switching from release to none?

**Comment by sgn** [comment]
> What's the reason for switching from release to none?

Release force `-O3`

File: srcpkgs/eigen/template [hunk 2]
```diff
@@ -3,6 +3,8 @@ pkgname=eigen
 version=3.3.9
 revision=1
 build_style=cmake
+# This is header only library, don't waste time to patch
```
Comment (Piraty) [inline srcpkgs/eigen/template:6]: this might make a good commit message, but not an in-source comment

**Comment by sgn** [comment]
Question: Should we add `-DNDEBUG` to `CFLAGS` and `CXXFLAGS`?

**Comment by ericonr** [comment]
I would rather not.

**Comment by ericonr** [comment]
Hmmm, unless CMake release mode adds it... Then I guess yes... This complicated.

**Comment by sgn** [comment]
> Hmmm, unless CMake release mode adds it... Then I guess yes... This complicated.

Yes, `-DNDEBUG` added in `Release` `MinSizeRel` and `RelWithDebInfo`.

**Comment by sgn** [comment]
`rspamd` vendored too much dependencies. I gave up on it.

**Comment by sgn** [comment]
Most patching are done.
Package that adds `-O2` and `-g` are ignored.
- `rspamd`, `yabause-gtk`, `yoshimi`, `zbackup`, `zig`, `zulucrypt` adds `-O3` but wasn't patched.

Would appreciate review from team.

**Comment by sgn** [comment]
@void-linux/pkg-committers please have a look. I expect to push this out next weekend if no major change required.

**Review by Piraty — APPROVED** [review]

**Pull request merged**